runtime usage

hansieodendaal opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
- Main `README.md` updated to better guide first-time users to run the base node.
- `wallet_identity_file` and `wallet_tor_identity_file` parameters added to `tari_config_sample.toml`
- All aliases for command-line arguments are now consistently supported across platforms; `--create-id` & `--create_id`, `--base-path` & `--base_path`, `base-dir` & `base_dir`, `log-config` & `log_config`.
- `--base-path` can now be specified as a relative path, previously an absolute path was required.
- `--base-path` can now be created multiple levels deep, previously only one new sub-level could be accommodated.
- Deprecated command-line args from matches methods and associated tests removed
- Tests expanded and added to cover all use cases.

Bug fixes:
- Consistent command-line arguments for Windows and Linux can now be applied, previously `--create-id` & `--create_id` was required respectively.
- Log config file was not created in the `--base-path` directory when the latter was specified.
- The log folder was not created in the `--base-path` directory when specified, but in the current working directory instead.

## Motivation and Context
- `README.md` did not give proper runtime guidance.
- Consistent command-line use between Windows and Linux is a requirement.
-  `tari_config_sample.toml` did not explain the wallet identities, with the result that uninformed users would not have unique base-node wallet identities if multiple base nodes were run on the same computer.

## How Has This Been Tested?
Tested on Windows and Linux

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
* [X] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [X] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [X] New Tests
* [ ] Documentation

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
* [X] I'm merging against the `development` branch.
* [X] I ran `cargo-fmt --all` before pushing.
* [X] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [X] I have added tests to cover my changes.

hansieodendaal commented:
@dunnock, could you please have a look at this one? Thanks

hansieodendaal commented:
Also fixed a bug with the log files location; it is now in the base-dir if no option is given. Previously it was always in the current dir.

hansieodendaal commented:
Something wrong with GitHub, will try to push tomorrow

hansieodendaal commented on common/src/configuration/bootstrap.rs:
```diff
@@ -132,7 +149,7 @@ impl ConfigBootstrap {
 
         let log_config = if self.log_config.to_str() == Some("") {
-            None
+            Some(dir_utils::default_path(DEFAULT_LOG_CONFIG, Some(&self.base_path)))
```
Actually no... `get_log_configuration_path` provides some additional options for specifying a log configuration file.

dunnock replied:
Though that functionality is activated only when `None` is supplied. 

I was looking at `default_path` impl:
```
        let mut home = dirs::home_dir().unwrap_or_else(|| PathBuf::from(".")); 
```
it does similar thing to `get_log_configuration_path`, considering that env var `TARI_LOG_CONFIGURATION` is already processed by StructOpt. The only difference in that case would be:
`default_path` fallback to current dir:
```
PathBuf::from(".")
```
vs `get_log_configuration_path` fallback to current dir:
```
env::current_dir().expect(
                "Could find a suitable path to the log configuration file. Consider setting the \
                 TARI_LOG_CONFIGURATION envar, or check that the current directory exists and that you have \
                 permission to read it",
            )
```

hansieodendaal replied:
@dunnock, I force pushed new code changes, please review the latest version.
Please suggest code changes for this comment if you still want me to change something.

dunnock replied:
@hansieodendaal I'm cool with all changes, it definitely makes code simpler to use. 
Except this small issue: `log_config` is always `Some`, hence `logging::get_log_configuration_path(log_config);` will always be only doing `.unwrap()` - why do we need that call now?

hansieodendaal replied:
Cool, I get you now, dit not realize that StructOpt actually parsed the environment variable. I removed the function `get_log_configuration_path`, removed tests that tested it, and expanded the `test_bootstrap_args_from_iter_safe` test to include parsing the environment variable.

hansieodendaal commented:
@dunnock, I force pushed new code changes, please review the latest version.
Please suggest code changes for this comment if you still want me to change something.

dunnock commented:
@hansieodendaal based on comment above please see below of what I mean, it should behave exactly as your code as `get_log_configuration_path` effectively is only doing `unwrap` there
https://github.com/tari-project/tari/compare/ho_runtime_use...dunnock:ho_runtime_use
if you agree perhaps we won't need `fn get_log_configuration_path` at all too

hansieodendaal commented:
@dunnock, cool, I get you now, dit not realize that StructOpt actually parsed the environment variable. I removed the function `get_log_configuration_path`, removed tests that tested it, and expanded the `test_bootstrap_args_from_iter_safe` test to include parsing the environment variable.

CjS77 commented on common/src/dir_utils.rs:
```diff
@@ -53,3 +54,68 @@ pub fn default_path(filename: &str, base_path: Option<&PathBuf>) -> PathBuf {
+    fn test_multiple_levels_create_data_directory() {
+        let temp_dir = TempDir::new(string(8).as_str()).unwrap();
+        let dir = &PathBuf::from(
+            temp_dir.path().to_path_buf().display().to_string().to_owned() +
+                "/" +
+                string(2).as_str() +
+                "/" +
+                string(2).as_str() +
+                "/" +
+                string(2).as_str() +
+                "/" +
+                string(2).as_str() +
+                "/" +
+                string(2).as_str() +
+                "/" +
+                string(2).as_str() +
+                "/" +
+                string(2).as_str() +
+                "/" +
+                string(2).as_str() +
+                "/" +
+                string(2).as_str() +
+                "/" +
+                string(2).as_str() +
+                "/" +
+                string(2).as_str() +
+                "/" +
+                string(2).as_str() +
+                "/",
+        );
```
You could do this in one line with something like
`[0..12].map(|_| string(2)).collect().join("/")`
